packages/react-core/src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Icon/icon';
+import { css } from '@patternfly/react-styles';
+
+export interface IconComponentProps extends React.HTMLProps<HTMLSpanElement> {
+  /** Icon content */
+  children?: React.ReactNode;
+  /** Progress icon content */
+  progressIcon?: React.ReactNode;
+  /** Additional classes applied to the icon container */
+  className?: string;
+  /** Size of the icon */
+  containerSize?: 'sm' | 'md' | 'lg' | 'xl';
+  /** Size of icon content */
+  contentSize?: 'sm' | 'md' | 'lg' | 'xl';
+  /** Status color of the icon */
+  status?: 'default' | 'info' | 'success' | 'warning' | 'danger';
+  /** Indicates the icon is inline */
+  isInline?: boolean;
+  /** Indicates the icon is in progress. Setting this property to true will hide any children, and make the progressIcon property visible. */
+  isInProgress?: boolean;
+}
+
+export const Icon: React.FunctionComponent<IconComponentProps> = ({
+  children,
+  progressIcon,
+  className,
+  containerSize,
+  contentSize,
+  status,
+  isInline = false,
+  isInProgress = false,
+  ...props
+}: IconComponentProps) => (
+  <span
+    className={css(
+      styles.icon,
+      isInline && styles.modifiers.inline,
+      isInProgress && styles.modifiers.inProgress,
+      styles.modifiers[containerSize],
+      className
+    )}
+    {...props}
+  >
+    <span className={css(styles.iconContent, styles.modifiers[contentSize], styles.modifiers[status])}>{children}</span>
+    {progressIcon && <span className={css(styles.iconProgress, className)}>{progressIcon}</span>}
```

Comment:
I wonder if we should include a `1em` spinner here by default, that a user can override if they pass something else

```
<Spinner isSVG diameter="1em" ... />
```

Reply:
I can add 1em diameter to the examples, the Icon component itself has no default for the progressIcon (should that be defaulting to a spinner?)

Reply:
FWIW @thatblindgeye added a [`pf-m-inline` variant](https://pf4.patternfly.org/components/spinner#inline) to the core spinner that sets the size to `1em` for use in [inline link buttons](https://pf4.patternfly.org/components/button#progress). That feature will need to be added to react before you can use it, but maybe you could use that instead of setting `diameter`?